table/tables/tables_test.go

```diff
@@ -397,6 +398,48 @@ PARTITION BY RANGE ( id ) (
 	tbInfo = tb.Meta()
 	_, err = tb.AddRecord(ts.se, types.MakeDatums(22), false)
 	c.Assert(err, IsNil) // Insert into maxvalue partition.
+
+	// test for hash partition table.
+	createTable1 = `CREATE TABLE test.t1 (id int(11), index(id)) PARTITION BY HASH (id) partitions 4; `
+	_, err = ts.se.Execute(context.Background(), "set @@session.tidb_enable_table_partition = 'on';")
```

Comment:
If `@@session.tidb_enable_table_partition` is off before `create table`, the created table is not a partitione table. @crazycs520